src/collection-view.js

```diff
@@ -282,10 +282,11 @@ Marionette.CollectionView = Marionette.AbstractView.extend({
   // Retrieve the `childView` class, either from `this.options.childView`
   // or from the `childView` in the object definition. The "options"
   // takes precedence.
-  // This method receives the model that will be passed to the instance
-  // created from this `childView`. Overriding methods may use the child
-  // to determine what `childView` class to return.
-  getChildView: function(child) {
+  // The `childView` property can be either a view class or a function that
+  // returns a view class. If it is a function, it will receive the model that
+  // will be passed to the view instance (created from the returned view class)
+
+  _getChildView: function(child) {
     var childView = this.getOption('childView');
 
     if (!childView) {
```

Comment:
I agree and I think we shouldn't have "custom errors".

On the other hand, for this case I actually think it is useful because I'd say one of the most common mistakes in using `CollectionView` is when the key "childview" is used instead of `childView`. Having a explicit message for this case can save some time.